src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/sort/SortExpression.java

```diff
@@ -18,7 +18,7 @@ public record SortExpression(
     boolean ascending,
     @Nullable float[] vector,
     @Nullable String vectorize,
-    @Nullable String bm25Query) {
+    @Nullable String lexicalQuery) {
```

Comment:
I made this comment elsewhere as well, now that we have better handle on how and when the SortExpression objects are created it feels like time to have either a typenum or a subclass structure. 

there are hidden behaviour in here: if the instance is created via the static factories than the is* functions work. But there is a public constructor that allows us to break all the rules.

Reply:
Yeah Records are just open structs (although immutable).